phoenix-core/src/it/java/org/apache/phoenix/end2end/CDCBaseIT.java

```diff
@@ -0,0 +1,655 @@
+                tableProps);
+        if (indexType != null && indexType == PTable.IndexType.LOCAL) {
+            props.add(TableProperty.INDEX_TYPE.getPropertyName() + "=" +
+                    (indexType == PTable.IndexType.LOCAL ? "l" : "g"));
+        }
+        if (nSaltBuckets != null) {
+            props.add(TableProperty.INDEX_TYPE.getPropertyName() + "=" + indexType);
+        }
+        Boolean immutableTable = (Boolean) TableProperty.IMMUTABLE_ROWS.getValue(tableProps);
+        if (immutableTable) {
+            props.add(TableProperty.IMMUTABLE_ROWS.getPropertyName() + "=true");
+        }
+        PTable.ImmutableStorageScheme immutableStorageScheme =
+                (PTable.ImmutableStorageScheme) TableProperty
+                        .IMMUTABLE_STORAGE_SCHEME.getValue(tableProps);
+        if (immutableStorageScheme != null) {
+            props.add(TableProperty.IMMUTABLE_STORAGE_SCHEME.getPropertyName() + "="
+                    + immutableStorageScheme.name());
+        }
+        table_sql += " " + String.join(", ", props);
+        conn.createStatement().execute(table_sql);
+    }
+
+    protected void createCDCAndWait(Connection conn, String tableName, String cdcName,
+                                    String cdc_sql) throws Exception {
+        createCDCAndWait(conn, tableName, cdcName, cdc_sql, null, null, null);
+    }
+
+    protected void createCDCAndWait(Connection conn, String tableName, String cdcName,
+                                  String cdc_sql, PTable.IndexType indexType) throws Exception{
+        createCDCAndWait(conn, tableName, cdcName, cdc_sql, null, null, indexType);
+    }
+
+    protected void createCDCAndWait(Connection conn, String tableName, String cdcName,
+                                    String cdc_sql, PTable.QualifierEncodingScheme encodingScheme,
+                                    Integer nSaltBuckets) throws Exception {
+        createCDCAndWait(conn, tableName, cdcName, cdc_sql, encodingScheme, nSaltBuckets, null);
+    }
+
+    protected void createCDCAndWait(Connection conn, String tableName, String cdcName,
+                                    String cdc_sql, PTable.QualifierEncodingScheme encodingScheme,
+                                    Integer nSaltBuckets, PTable.IndexType indexType) throws Exception {
+        // For CDC, multitenancy gets derived automatically via the parent table.
+        createTable(conn, cdc_sql, encodingScheme, false, nSaltBuckets, indexType, false, null);
+        String schemaName = SchemaUtil.getSchemaNameFromFullName(tableName);
+        tableName = SchemaUtil.getTableNameFromFullName(tableName);
+        IndexToolIT.runIndexTool(false, schemaName, tableName,
+                "\""+CDCUtil.getCDCIndexName(cdcName)+"\"");
+        String indexFullName = SchemaUtil.getTableName(schemaName,
+                CDCUtil.getCDCIndexName(cdcName));
+        TestUtil.waitForIndexState(conn, indexFullName, PIndexState.ACTIVE);
```

Comment:
Is this necessary? IndexTool runs and waits for the job to complete.

Reply:
This was copied from existing test code (e.g, see [1](https://github.com/apache/phoenix/blob/1e96a2756eaf0a2201a50579789190e8c10747df/phoenix-core/src/it/java/org/apache/phoenix/end2end/index/BaseIndexIT.java#L1516) [2](https://github.com/apache/phoenix/blob/1e96a2756eaf0a2201a50579789190e8c10747df/phoenix-core/src/it/java/org/apache/phoenix/end2end/index/BaseIndexWithRegionMovesIT.java#L1711)). But now, I see plenty of other places where this wasn't used, so I will try without and remove if it is working fine.